guides/keras_cv/retina_net_overview.py

```diff
@@ -0,0 +1,275 @@
+the model are expected to be in the range `[0, 255]`.
+"""
+
+model = keras_cv.models.RetinaNet(
+    classes=20,
+    bounding_box_format="xywh",
+    backbone="resnet50",
+    backbone_weights="imagenet",
+    include_rescaling=True,
+)
+
+"""
+That is all it takes to construct a KerasCV RetinaNet.  The RetinaNet accepts tuples of
+dense image Tensors and ragged bounding box Tensors to `fit()` and `train_on_batch()`
+This matches what we have constructed in our input pipeline above.
+
+The RetinaNet `call()` method outputs two values: training targets and inference targets.
+In this guide, we are primarily concerned with the inference targets.  Internally, the
+training targets are used by `keras_cv.losses.ObjectDetectionLoss()` to train the
+network.
+"""
+
+"""
+## Optimizer
+
+For training, we use a SGD optimizer with a piece-wise learning rate schedule
+consisting of a warm up followed by a ramp up, then a ramp.
+Below, we construct this using a `keras.optimizers.schedules.PiecewiseConstantDecay`
+schedule.
+"""
+
+optimizer = optimizers.SGD(learning_rate=0.1, momentum=0.9, global_clipnorm=10.0)
+
+"""
+## COCO metrics monitoring
+
+KerasCV offers a suite of in-graph COCO metrics that support batch-wise evaluation.
+More information on these metrics is available in:
+
+- [Efficient Graph-Friendly COCO Metric Computation for Train-Time Model Evaluation](https://arxiv.org/abs/2207.12120)
+- [Using KerasCV COCO Metrics](https://keras.io/guides/keras_cv/coco_metrics/)
+
+Lets construct two COCO metrics, an instance of
+`keras_cv.metrics.COCOMeanAveragePrecision` with the parameterization to match the
+standard COCO Mean Average Precision metric, and `keras_cv.metrics.COCORecall`
+parameterized to match the standard COCO Recall metric.
+"""
+
+metrics = [
+    keras_cv.metrics.COCOMeanAveragePrecision(
+        class_ids=range(20),
+        bounding_box_format="xywh",
+        name="Mean Average Precision",
+    ),
+    keras_cv.metrics.COCORecall(
+        class_ids=range(20),
+        bounding_box_format="xywh",
+        max_detections=100,
+        name="Recall",
+    ),
+]
+
+"""
+## Training our model
+
+All that is left to do is train our model.  KerasCV object detection models follow the
+standard Keras workflow, leveraging `compile()` and `fit()`.
+
+Let's compile our model:
+"""
+
+loss = keras_cv.losses.ObjectDetectionLoss(
+    classes=20,
+    classification_loss=keras_cv.losses.FocalLoss(from_logits=True, reduction="none"),
+    box_loss=keras_cv.losses.SmoothL1Loss(l1_cutoff=1.0, reduction="none"),
+    reduction="auto",
+)
+
+model.compile(
+    loss=loss,
+    optimizer=optimizer,
+    metrics=metrics,
+)
+
+"""
+All that is left to do is construct some callbacks:
+"""
+
+callbacks = [
+    callbacks_lib.TensorBoard(log_dir="logs"),
+    callbacks_lib.EarlyStopping(patience=50),
+    callbacks_lib.ReduceLROnPlateau(patience=20),
+]
+
+"""
+And run `model.fit()`!
+"""
+
+model.fit(
```

Comment:
We should add an inference section as well, including visualizing inference results (we could just drop saved images of the final results in the markdown since the model here is not trained to convergence)

Reply:
Sounds good, once we finish up stabilizing the API I'll run this on Colab and generate some nice visuals.